### PR TITLE
update istio to version 1.12.0

### PR DESCRIPTION
**Pull request by DockToFuture**
**How to categorize this PR?**

/area networking
/kind enhancement

**What this PR does / why we need it**:
Update istio to version `1.12.0` since support for istio version `1.10.x` is dropped by the end of year. See: https://istio.io/latest/docs/releases/supported-releases/.
Additionally clean up old envoyfilter which are no longer required.

**Special notes for your reviewer**:

The metadata exchange server are no longer required in istio `1.12.0`

**Release note**:

```other operator
Update istio to version `1.12.0`.
```

### CODE REVIEW

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Please update the envoy filters as well.

**Review by ScheererJ — CHANGES_REQUESTED** [review]
Please remove the metadata envoy filter as discussed as they are natively implemented in istio 1.12.

**Review by ScheererJ — DISMISSED** [review]
/lgtm

**Review by rfranzke — DISMISSED** [review]
/lgtm

File: charts/images.yaml [hunk 1]
```diff
@@ -278,11 +278,11 @@ images:
 - name: istio-proxy
   sourceRepository: github.com/istio/istio
   repository: gcr.io/istio-release/proxyv2
-  tag: "1.10.2-distroless"
+  tag: "1.12.0-distroless"
```
Comment (ialidzhikov) [inline charts/images.yaml:281]: Do you plan also to update to go.mod dependency to 1.12.0? When I check previous upgrade PRs, both image version and go.mod dependency version were updated in a single PR - for example https://github.com/gardener/gardener/pull/4241.

Comment (ialidzhikov) [inline charts/images.yaml:281]: Isn't there any change required to the istio CRDs (`./charts/istio/istio-crds/templates/crd-all.gen.yaml`)? I can see that previous upgrade PRs were actively modifying/synching this 5k lines long CRDs file. Can you explain how this CRDs file is maintained? Is it a copy of https://github.com/istio/istio/blob/1.12.0/manifests/charts/base/crds/crd-all.gen.yaml?

Comment (DockToFuture) [reply charts/images.yaml:281]: Thank you for the hint, updated it accordingly.

Comment (DockToFuture) [reply charts/images.yaml:281]: It's a copy for the file at least it was for the last update. In the initial commit were also some crds removed which no longer exist.

**Review by ScheererJ — APPROVED** [review]
/lgtm

**Comment by rfranzke** [comment]
@ialidzhikov Do you have further comments?

**Comment by ialidzhikov** [comment]
> @ialidzhikov Do you have further comments?

Let me check again

**Review by ialidzhikov — APPROVED** [review]
/lgtm
/needs second-opinion

**Review by rfranzke — APPROVED** [review]
/lgtm
/unhold

**Pull request merged**